server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -682,7 +686,7 @@ private OsStats.Cgroup getCgroup() {
             assert controllerMap.isEmpty() == false;
 
             final String cpuAcctControlGroup;
-            final long cgroupCpuAcctUsageNanos;
+            final String cgroupCpuAcctUsageNanos;
```

Comment:
Is it a good choice to keep this as a String, if we need to go back and forth? (I see at least a couple of `toString()` here and there)

Reply:
Do you suggest to keep this as BigInteger? Then only in writeTo and in constructor we will need to do the toString.

Reply:
You probably won't need that - StreamOutput and StreamInput support BigInteger AFAIK

Reply:
StreamOutput does not explicitly has a writeBigInteger method. However, 
1. It can be converted to byte array and then written.
2. It supports a writeGenericValue method which can take any object. While reading back we need to similarly read it using readGenericValue method. This will lead to a casting to BigInteger class since the readGenericValue returns an object.

It might be better to add writeBigInteger method in StreamOutput.java and do a toString there. StreamInput.java already has a readBigInteger method which uses string to get the BigInteger.